### ISSUE TITLE
HeXen II - Playthrough Notes

### ISSUE THREAD

**Issue by eukara** [fte-team/fteqw#15]
Besides the bugs listed [here](https://github.com/fte-team/fteqw/issues/5)...
Other observations by eqagunn who played through the entire game (except the missionpack):

- [ ]     cl_bob, cl_rollangle, and vid_conautoscale not in options.
- [ ]     Diagonal speed correction not in options. ([referring to a hack I did for Q/H2 you can find here](http://icculus.org/~marco/files/csqc-movementtest-run.zip))
- [ ]     New Mission music loads for Old Mission. (when installed into the data dir)
- [ ]     Z-fighting present throughout the game. (cvars need tweaking specifically for HeXen II)
- [ ]     Messages can't be centered. (and ezHud plugin is incompatible, so there ought to be an option)
- [ ]     Currently selected item not shown on mini HUD.
- [ ]     Savegames can be cross-loaded and get corrupted. (that is a technical limitation with precache lists, although we might have to simply disallow loading of savegames with non-existant resources unless forced)

**Comment by eqagunn** [comment]
With Portal of Praevus completed, adding to the list:
* [ ] [sv_stepheight=137](https://i.imgur.com/1S82QtC.jpeg) and [sv_stepheight=113](url) required in map MESO2, and [another one](https://i.imgur.com/SBlebBj.jpeg) in KEEP1 with sv_stepheight having no effect on it, but jumping over does.
* [ ] One thing about mouse and menus in fteqw. Mouse usage would feel better if hovering was also selecting. At the moment you have to click to select, but clicking also alters values. So for example if you want to select with mouse and fine tune with keyboard, you can't.
* [ ] If standing on a slope player slides down slowly instead of staying in the same spot.
* [ ] The game should launch without the expansion if `-portals` is not used. One should switch to the expansion from mods. Currently it launches with it regardless if `-portals` is used or not.
* [ ] More often then not, when character changes map, the sound for resurfacing breathing is being played.
* [ ] Controls have Jump/SwimUp and standalone SwimUp. Didn't have the latter bound as I thought I don't need a duplicate key for the same action, but came to realize that Jump/SwimUp cannot take you out of some pools with higher edge, while SwimUp can. Therefore, it would be beneficial to have the option to make Jump execute both `+jump` and `+moveup`, and Crouch both `+crouch` and `+movedown`. This would make diving and flying more intuitive.
* [ ] mid-screen gameplay messages stay on for only 2 sec. Not enough to read them. For pretty much all of them I had to reopen console to see what it was. 5 sec would fix that.

**Comment by Xylemon** [comment]
Merging some issues over from the SourceForge tracker I thought should be here.

**Comment by Xylemon** [comment]
- [ ] side speed is wrong

https://sourceforge.net/p/fteqw/tickets/135/

bublegum wrote on 2022-03-17:

Version: 6202

In regular Hexen II and Hexen II specific source ports, cl_sidespeed is entirely ignored. Instead, the value of cl_forwardspeed is used (400) for side step.
FTEQW though does not ignore the value of cl_sidespeed (225), thus it is way too slow, which is immediately apparent when strafing.

This can be trivially worked-around by users by setting cl_sidespeed to 400. But it would be better if it worked correctly out of the box. For this, if FTEQW detects it is running Hexen II, it should automatically internally use the value of cl_forwardspeed for both cl_sidespeed and cl_backspeed: Hexen II uses a single speed for all directions (400 by default).

---

@Shpoike wrote on 2022-04-10:

Actually that's not quite right.
The h2mp code uses cl_forwardspeed>200 ONLY as an always-run flag - set cl_movespeedkey to 1 and you should see that it stops having any effect at all.
Instead its actual forwardspeed is hardcoded to 200, and its sidespeed is hardcoded to 225.

FTE's default cl_sidespeed is already 400 (should actually be 450 to match hexen2), probably you're only noticing because of existing configs (probably including default.cfg...)
I can make FTE's options menu force them appropriately when running hexen2 (but setting cl_run instead of cl_forwardsped>200), but that won't help existing configs until they fiddle with the menus.

The speed-boots artifact is also an issue however. FTE is lazy and basically just scales maxspeed without scaling intents, while hexen2 has a higher maxspeed and scales intents instead.

**Comment by Xylemon** [comment]
- [ ] bodies on the ground cannot be gibbed

https://sourceforge.net/p/fteqw/tickets/136/

bublegum wrote on 2022-03-17:

Version: 6202

Unlike regular Hexen and Hexen 2 specific source port, it is not possible in FTEQW to gib dead bodies on the ground. For example, on the startup level, you cannot gib a dead archer.

**Comment by Xylemon** [comment]
- [ ] AI pathing issue on meso2

https://sourceforge.net/p/fteqw/tickets/137/

At the beginning of map meso2, there is a weird issue with AI pathing, where AI suicides in the lava.
This does not happen in regular Hexen 2 nor Hexen 2 specific source ports.
See the attached demo (to reprod: skill 2; map meso2) with both scorpions and the golem going into lava.

**1 Attachments**

[Player_meso2.qwd.gz](https://github.com/fte-team/fteqw/files/10318844/Player_meso2.qwd.gz)

**Comment by Xylemon** [comment]
- [ ] Hexen 2: issue with moving platform on meso2

https://sourceforge.net/p/fteqw/tickets/138/

bublegum wrote on 2022-03-17:

At the start of map meso2, there is a golem standing on a high platform that automatically lowers.
Once the platform is at ground level, it is not possible to go over the platform without jumping. It's also the case for a bridge over lava that appears after enabling a switch.
This is not the case in regular Hexen 2 and Hexen 2 specific source ports.
The second part of the attached demo shows the problem.

**1 Attachments**

[Player_meso2.qwd.gz](https://github.com/fte-team/fteqw/files/10318844/Player_meso2.qwd.gz)

**Comment by Xylemon** [comment]
Not posting https://sourceforge.net/p/fteqw/tickets/96/ as those issues are already documented here.

**Comment by washingleavesandthensome23-434** [comment]
Any progress on these issues? Including transparency issues mentioned?🙂

**Comment by nashmuhandes** [comment]
I suppose this is the Hexen 2 megathread? I posted these on Discord before, but I think it would be better to post these on a proper issue tracker.

Issues for both Hexen 2 and Portals:

- [ ] The "drop item" command does not work
- [ ] Cannot use melee weapons to gib enemy corpses
- [ ] The autorun option in the menu doesn't seem to work - you must hold down the `+speed` key to run
- [ ] I'll need to reconfirm this, but it seems that running speed (when holding down `+speed`) seems WAY too fast?
- [ ] The startup demos refuse to play (even if force-enabled with `cl_demoreel 1`)
- [ ] snd_musicvolume seems to default to 0, misleading users to think music does not work
- [ ] Playing with both VSync turned off and unlimited frame rate (by setting it to 0 in the FPS menu) makes monsters unkillable (probably shouldn't play with unlimited frame rate anyway...)
- [ ] The stats/inventory tab doesn't have the fancy ease-in/out animation and doesn't play the sound assigned to it
- [ ] Ring graphics in the stats tab are corrupted ("unsupported image format" error)
- [ ] No selective framegroup no-lerps - torches/fire look bad when lerped, while the waving flags look good, for example
- [ ] Almost no dynamic lights at all from any projectiles
- [ ] Player's lower color is initially garbled/corrupted
- [ ] Inventory select and use sounds missing

Portals-specific issues:

- [ ] The machinima-style intro cinematic is missing
- [ ] Some text-based intermissions don't work
- [ ] The "Objective" system is only shown on the screen as a top-left message, rather than drawn with a fancy plaque graphic in the middle of the screen
- [ ] You cannot have separate OGG music for the base game ("old mission") and Portals ("new mission") because FTE only finds music files named `trackxx`, instead of the uniquely-named tracks of the OG game (`casa1`, `tulku7`, etc). What happens is that the music folder of `Portals` will override the ones in the `data1` folder. Naming the OGGs from `trackxx` to their uniquely named equivalents simply won't work.

**Comment by bubbleguuum** [comment]
Hexen 2 specific commits from 2 weeks ago fixed many issues I reported:

- inability to gib dead bodies on the ground
- on meso2, the golem and 2 spiders at the start of the level going suicidal into the lava
- strafe speed being too slow
- with always run enabled, using the run key not resulting in walk speed

Many thank to @Shpoike for giving Hexen 2 some much needed attention, making fteqw more viable for playing that game.
This is especially important since fteqw is the only Hexen 2 source port offering some specific features such as model interpolation. Since we may never get an official remaster for Hexen 2, source ports are really important.

**Comment by nashmuhandes** [comment]
> Hexen 2 specific commits from 2 weeks ago fixed many issues I reported (tested on a fresh git compile on Linux):
> 
> * inability to gib dead bodies on the ground
> * on meso2, the golem and 2 spiders at the start of the level going suicidal into the lava
> * strafe speed being too slow
> * with always run enabled, using the run key not resulting in walk speed
> 
> Many thanks to @Shpoike for giving Hexen 2 some much needed attention, making fteqw more viable for playing that game in a proper way. This is especially important since fteqw is the only Hexen 2 source port offering some specific features such as model interpolation. Since we may never get an official remaster for Hexen 2, source ports are really important.

Agreed. As nice as the _other_ Hexen 2 port is (you know the one), it's still missing a lot of modern QoL and niceties. And, more importantly for me and my friendgroup, it's unplayable for internet multiplayer across countries. FTEQW's netcode with the player prediction stuff is the only proper way for us to play some Hexen 2 multiplayer. We hope to do another coop H2 playthrough with FTE now that a lot of the issues are fixed.

**Comment by legluondunet** [comment]
I confirm that this issues:
